validator/db/kv/proposer_settings.go

```diff
@@ -0,0 +1,132 @@
+		b := bkt.Get(proposerSettingsKey)
+		if len(b) == 0 {
+			return fmt.Errorf("no proposer settings found in bucket")
+		}
+		to := &validatorpb.ProposerSettingsPayload{}
+		if err := proto.Unmarshal(b, to); err != nil {
+			return errors.Wrap(err, "failed to unmarshal proposer settings")
+		}
+		settings, err := validatorServiceConfig.ToSettings(to)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert payload to proposer settings")
+		}
+		if settings.ProposeConfig == nil {
+			settings.ProposeConfig = make(map[[fieldparams.BLSPubkeyLength]byte]*validatorServiceConfig.ProposerOption)
+		}
+		settings.ProposeConfig[pubkey] = options
+		m, err := proto.Marshal(settings.ToPayload())
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal proposer settings")
+		}
+		return bkt.Put(proposerSettingsKey, m)
+	})
+	return err
+}
+
+// UpdateProposerSettingsDefault updates the existing default settings for proposer settings
+func (s *Store) UpdateProposerSettingsDefault(ctx context.Context, options *validatorServiceConfig.ProposerOption) error {
+	_, span := trace.StartSpan(ctx, "validator.db.UpdateProposerSettingsDefault")
+	defer span.End()
+	if options == nil {
+		return errors.New("proposer settings option was empty")
+	}
+	if options.FeeRecipientConfig == nil {
+		return errors.New("fee recipient cannot be empty")
+	}
+	err := s.db.Update(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(proposerSettingsBucket)
+		b := bkt.Get(proposerSettingsKey)
+		if len(b) == 0 {
+			return NoProposerSettingsFound
+		}
+		to := &validatorpb.ProposerSettingsPayload{}
+		if err := proto.Unmarshal(b, to); err != nil {
+			return errors.Wrap(err, "failed to unmarshal proposer settings")
+		}
+		settings, err := validatorServiceConfig.ToSettings(to)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert payload to proposer settings")
+		}
+		settings.DefaultConfig = options
+		m, err := proto.Marshal(settings.ToPayload())
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal proposer settings")
+		}
+		return bkt.Put(proposerSettingsKey, m)
+	})
+	return err
+}
+
+// ProposerSettings gets the current proposer settings
+func (s *Store) ProposerSettings(ctx context.Context) (*validatorServiceConfig.ProposerSettings, error) {
+	_, span := trace.StartSpan(ctx, "validator.db.ProposerSettings")
+	defer span.End()
+	to := &validatorpb.ProposerSettingsPayload{}
+	if err := s.db.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(proposerSettingsBucket)
+		b := bkt.Get(proposerSettingsKey)
+		if len(b) == 0 {
+			return NoProposerSettingsFound
+		}
+		if err := proto.Unmarshal(b, to); err != nil {
+			return errors.Wrap(err, "failed to unmarshal proposer settings")
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	return validatorServiceConfig.ToSettings(to)
+}
+
+// ProposerSettingsExists returns true or false if the settings exist or not
+func (s *Store) ProposerSettingsExists(ctx context.Context) (bool, error) {
```

Comment:
Unit test?

Reply:
![image](https://github.com/prysmaticlabs/prysm/assets/90280386/7a45eda5-9c98-43f8-95de-88b77500603d)
mostly covered by the node_test.go will have other tests to cover related.